client/components/common/proptypes/index.js

```diff
@@ -0,0 +1,322 @@
+import PropTypes from 'prop-types';
```

Comment:
Wow, huge plus for putting this together and making all those changes. No more having to figure out how to account for certain structures :).

Also appreciate the care taken for identifying what should be required per prop type.